src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/adjust_brgemm_copy_b_loop_ports.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/itt.hpp"
+
+#include "adjust_brgemm_copy_b_loop_ports.hpp"
+#include "snippets/utils/utils.hpp"
+#include "snippets/lowered/loop_manager.hpp"
+#include "snippets/lowered/expressions/buffer_expression.hpp"
+#include "transformations/snippets/x64/op/brgemm_copy_b.hpp"
+#include "transformations/snippets/x64/op/brgemm_cpu.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+bool pass::AdjustBrgemmCopyBLoopPorts::update_loop_info(const std::shared_ptr<snippets::lowered::UnifiedLoopInfo>& loop_info) {
+    OPENVINO_ASSERT(loop_info, "Invalid loop info pointer");
+    bool modified = false;
+    auto caller = [&](snippets::lowered::LoopPort &loop_port,
+                      snippets::lowered::UnifiedLoopInfo::LoopPortDesc &loop_desc) {
+        const auto& p = *loop_port.expr_port;
+        if (p.get_type() == snippets::lowered::ExpressionPort::Input &&
+            p.get_index() == 1) {
+            const auto& node = p.get_expr()->get_node();
+            if (auto brg = as_type_ptr<BrgemmCPU>(node)) {
+                const auto precision = node->get_input_element_type(1);
+                if (brgemm_utils::with_repacking(brg->get_type()) &&
+                    precision != element::f32 && loop_port.is_incremented) {
```

Comment:
I see that in runtime we always iterate through all stored `affected` loops (in `RuntimeConfigurator::update`) and then we check if the current `affected` loop is really needed to be updated (non fp32, loop port in incremented and there is copy b).

What's about to store really needed Loops (already covered by these conditions) in the method `run()` below using the same checks there? If we store needed loops only once, we shouldn't check conditions one more time in the method `update_loop_info` which is called in runtime.
Moreover, if I call `update_loop_info`, I expect that this method will really update my LoopInfo. However, current implementation can return `false` and not update in some cases.

Reply:
We already only store loops that satisfy these conditions: a loop is not added to `m_affected_loops` if it was not updated [code](https://github.com/IvanNovoselov/openvino/blob/08360b6d9d6aaab23697b6ef4a0c2482ae4adde1/src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/adjust_brgemm_copy_b_loop_ports.cpp#L96-L98)

In my opinion, the only optimization we could make here is to store affected ports only (because not all ports of affected loops are affected obviously). I tried to do it, but it would significantly complicate the code without appreciable performance benefit. The main problem with this approach is that our loop infrastructure was not designed to work with loop ports directly, the lowest granularity we can conveniently work with is loop info. So accessing loop ports (and descriptors) directly would be quite hacky, and it doesn't make much sense since it's a temporary solution anyway. I would rather focus on blocked shapes instead.

Reply:
Oh I see that I mixed up `Loop` with `LoopPort`. For sure, then the current solution is ok. Thanks for the explanation!